src/main/java/com/onehouse/storage/S3AsyncStorageClient.java

```diff
@@ -100,25 +98,34 @@ private List<File> processListObjectsV2Response(ListObjectsV2Response response,
   }
 
   @Override
-  public CompletableFuture<InputStream> readFileAsInputStream(String s3Uri) {
-    return readFileFromS3(s3Uri).thenApplyAsync(BytesWrapper::asInputStream);
+  public CompletableFuture<FileStreamData> streamFileAsync(String s3Uri) {
+    log.debug("Reading S3 file as InputStream: {}", s3Uri);
+    GetObjectRequest getObjectRequest = getObjectRequest(s3Uri);
+    return s3AsyncClientProvider
+        .getS3AsyncClient()
+        .getObject(getObjectRequest, AsyncResponseTransformer.toBlockingInputStream())
```

Comment:
do we have a ticket for addressing memory utilization in controller? if not, we can create one & implement this.

Reply:
created a ticket https://app.clickup.com/t/86duetnq8